apps/server/src/modules/tool/external-tool/service/external-tool.service.ts

```diff
@@ -148,4 +154,20 @@ export class ExternalToolService {
 		config.redirectUris = oauthClient.redirect_uris;
 		config.frontchannelLogoutUri = oauthClient.frontchannel_logout_uri;
 	}
+
+	async fetchBase64Logo(logoUrl: string | undefined): Promise<string | null> {
```

Comment:
Allowing the `logoUrl` to be undefined makes no sense. Either it is defined or you don't call this function.